prov/shm/src/smr.h

```diff
@@ -225,7 +233,15 @@ void smr_format_iov(struct smr_cmd *cmd, fi_addr_t peer_id,
 		const struct iovec *iov, size_t count, size_t total_len,
 		uint32_t op, uint64_t tag, uint64_t data, uint64_t op_flags,
 		void *context, struct smr_region *smr, struct smr_resp *resp,
-		struct smr_cmd *pend);
+		struct smr_tx_entry *pend);
+void smr_format_offset(struct smr_cmd *cmd, fi_addr_t peer_id,
+		const struct iovec *iov, size_t count, size_t total_len,
+		uint32_t op, uint64_t tag, uint8_t datatype,
+		uint8_t atomic_op, uint64_t data, uint64_t op_flags,
+		void *context, struct smr_region *smr,
+		struct smr_region *peer_smr, struct smr_inject_buf **tx_bufs,
+		uint64_t buf_count, struct smr_resp *resp,
+		struct smr_tx_entry *pend);
```

Comment:
This is an impressive number of arguments. Maybe it would be cleaner to make use of 2-3 format calls that the caller could invoke?